tests/modules/ejector/test_ejector.py

```diff
@@ -0,0 +1,460 @@
+        with monkeypatch.context() as m:
+            m.setattr(
+                ejector_module,
+                "is_active_validator",
+                Mock(side_effect=lambda v, _: bool(v)),
+            )
+            yield
+
+    @pytest.mark.unit
+    @pytest.mark.usefixtures("consensus_client")
+    def test_get_churn_limit_no_validators(self, ejector: Ejector) -> None:
+        ejector.w3.cc.get_validators = Mock(return_value=[])
+        result = ejector._get_churn_limit(ref_blockstamp)
+        assert (
+            result == ejector_module.MIN_PER_EPOCH_CHURN_LIMIT
+        ), "Unexpected churn limit"
+
+    @pytest.mark.unit
+    @pytest.mark.usefixtures("consensus_client")
+    def test_get_churn_limit_validators_less_than_min_churn(
+        self,
+        ejector: Ejector,
+        ref_blockstamp: ReferenceBlockStamp,
+        monkeypatch: pytest.MonkeyPatch,
+    ) -> None:
+        with monkeypatch.context() as m:
+            ejector.w3.cc.get_validators = Mock(return_value=[1, 1, 0])
+            m.setattr(ejector_module, "MIN_PER_EPOCH_CHURN_LIMIT", 4)
+            m.setattr(ejector_module, "CHURN_LIMIT_QUOTIENT", 1)
+            result = ejector._get_churn_limit(ref_blockstamp)
+            assert result == 4, "Unexpected churn limit"
+            ejector.w3.cc.get_validators.assert_called_once_with(ref_blockstamp)
+
+    @pytest.mark.unit
+    @pytest.mark.usefixtures("consensus_client")
+    def test_get_churn_limit_basic(
+        self,
+        ejector: Ejector,
+        ref_blockstamp: ReferenceBlockStamp,
+        monkeypatch: pytest.MonkeyPatch,
+    ) -> None:
+        with monkeypatch.context() as m:
+            ejector.w3.cc.get_validators = Mock(return_value=[1] * 99)
+            m.setattr(ejector_module, "MIN_PER_EPOCH_CHURN_LIMIT", 0)
+            m.setattr(ejector_module, "CHURN_LIMIT_QUOTIENT", 2)
+            result = ejector._get_churn_limit(ref_blockstamp)
+            assert result == 49, "Unexpected churn limit"
+            ejector._get_churn_limit(ref_blockstamp)
+            ejector.w3.cc.get_validators.assert_called_once_with(ref_blockstamp)
+
+
+@pytest.mark.unit
+def test_get_processing_state(ejector: Ejector, blockstamp: BlockStamp) -> None:
+    result = ejector._get_processing_state(blockstamp)
+    assert isinstance(
+        result, EjectorProcessingState
+    ), "Unexpected processing state response"
+
+
+@pytest.mark.unit
+@pytest.mark.usefixtures("consensus_client")
+def test_get_latest_exit_epoch(
+    ejector: Ejector, blockstamp: BlockStamp, monkeypatch: pytest.MonkeyPatch
+) -> None:
+    ejector.w3.cc.get_validators = Mock(
+        return_value=[
+            Mock(validator=Mock(exit_epoch=999)),
+            Mock(validator=Mock(exit_epoch=42)),
+            Mock(validator=Mock(exit_epoch=42)),
+            Mock(validator=Mock(exit_epoch=1)),
+        ]
+    )
+
+    with monkeypatch.context() as m:
+        m.setattr(ejector_module, "FAR_FUTURE_EPOCH", 999)
```

Comment:
You can pass FAR_FUTURE_EPOCH as an exit_epoch for the first validator. Its more obvious than attribute mocking

Reply:
Nice catch, fixed.